enterprise_subsidy/apps/transaction/management/commands/backpopulate_transaction_email_and_title.py

```diff
@@ -0,0 +1,76 @@
+"""
+Management command to backpopulate transaction email and title
+"""
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+from openedx_ledger.models import Transaction
+
+from enterprise_subsidy.apps.subsidy.models import Subsidy
+from enterprise_subsidy.apps.transaction.utils import batch_by_pk
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Management command for backpopulating transaction email and title
+
+    ./manage.py backpopulate_transaction_email_and_title
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.dry_run = False
+
+    def add_arguments(self, parser):
+        """
+        Entry point for subclassed commands to add custom arguments.
+        """
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            help=(
+                'If set, no updates will occur; will instead log '
+                'the actions that would have been taken.'
+            ),
+        )
+
+    def process_transaction(self, subsidy, txn):
+        """
+        Given a transaction (and it's subsidy), backpopulate the email and title
+        """
+        logger.info(f"Processing {subsidy.uuid} transaction {txn.uuid}")
+
+        if txn.lms_user_email is None:
+            lms_user_email = subsidy.email_for_learner(txn.lms_user_id)
```

Comment:
Do we want to make these calls if `self.dry_run` is true?  I could go either way - on the one hand, it's nice to set `--dry-run` _just_ to see what would be updated, and then have that happen relatively quickly.  On the other hand, it's nice to know with a dry run if we're actually fetching the data we expect.

Reply:
i think we wanna be able to see that these API calls work without actually saving the results